controllers/ovs/flowsconfig.go

```diff
@@ -0,0 +1,144 @@
+package ovs
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/netobserv/network-observability-operator/controllers/constants"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	flowsv1alpha1 "github.com/netobserv/network-observability-operator/api/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	ovsConfigName = "ovs-flows-config"
+)
+
+type FlowsConfigController struct {
+	operatorNamespace string
+	client            client.Client
+}
+
+func NewFlowsConfigController(client client.Client, operatorNamespace string) FlowsConfigController {
+	return FlowsConfigController{
+		client:            client,
+		operatorNamespace: operatorNamespace,
+	}
+}
+
+// Reconcile reconciles the status of the ovs-flows-config configmap with
+// the target FlowCollector ipfix section map
+func (c *FlowsConfigController) Reconcile(
+	ctx context.Context, target *flowsv1alpha1.FlowCollector) error {
+	rlog := log.FromContext(ctx, "component", "FlowsConfigController")
+
+	if !target.ObjectMeta.DeletionTimestamp.IsZero() {
+		rlog.Info("no need to reconcile status of a FlowCollector that is being deleted. Ignoring")
+		return nil
+	}
+	current, err := c.current(ctx)
+	if err != nil {
+		return err
+	}
+	desired, err := c.desired(ctx, target)
+	// compare current and desired
+	if err != nil {
+		return err
+	}
+
+	if current == nil {
+		rlog.Info("Provided IPFIX configuration. Creating " + ovsConfigName + " ConfigMap")
+		return c.client.Create(ctx, c.flowsConfigMap(desired))
+	}
+
+	if desired != current {
+		rlog.Info("Provided IPFIX configuration differs current configuration. Updating")
+		return c.client.Update(ctx, c.flowsConfigMap(desired))
+	}
+
+	rlog.Info("No changes needed")
+	return nil
+}
+
+func (c *FlowsConfigController) Finalize(ctx context.Context) error {
+	err := c.client.DeleteAllOf(ctx, &corev1.ConfigMap{
```

Comment:
what's the difference with a simple call of `client.Delete` ? we're deleting a single object from its name, no?

Reply:
Delete just works. Changed